src/valid/expression.rs

```diff
@@ -384,6 +384,14 @@ impl super::Validator {
                         kind: crate::ScalarKind::Float,
                         multi: false,
                     } => false,
+                    crate::ImageClass::Sampled {
+                        kind: crate::ScalarKind::Uint,
+                        multi: false,
+                    } => false,
+                    crate::ImageClass::Sampled {
+                        kind: crate::ScalarKind::Sint,
+                        multi: false,
+                    } => false,
```

Comment:
```suggestion
                    crate::ImageClass::Sampled {
                        kind: crate::ScalarKind::Uint | crate::ScalarKind::Sint,
                        multi: false,
                    } if gather.is_some() => false,
```

We should only allow these if it's a gather operation.